src/Provider/HttpKernelServiceProvider.php

```diff
@@ -0,0 +1,98 @@
+use DI\Bridge\Silex\ConverterListener;
+use DI\Bridge\Silex\MiddlewareListener;
+use Interop\Container\ContainerInterface;
+use Invoker\ParameterResolver\AssociativeArrayResolver;
+use Invoker\ParameterResolver\Container\TypeHintContainerResolver;
+use Invoker\ParameterResolver\ResolverChain;
+use Pimple\ServiceProviderInterface;
+use Silex\Api\EventListenerProviderInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @author Jacob Dreesen <jacob.dreesen@gmail.com>
+ */
+class HttpKernelServiceProvider implements ServiceProviderInterface, EventListenerProviderInterface
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @var CallbackInvoker
+     */
+    private $callbackInvoker;
+
+    /**
+     * @param ContainerInterface $container
+     * @param CallbackInvoker    $callbackInvoker
+     */
+    public function __construct(ContainerInterface $container, CallbackInvoker $callbackInvoker)
+    {
+        $this->container = $container;
+        $this->callbackInvoker = $callbackInvoker;
+    }
+
+    public function register(\Pimple\Container $app)
+    {
+        // Override the controller resolver with ours.
+        $app['resolver'] = function ($app) {
+            return new ControllerResolver(
+                $app['phpdi.callable_resolver'],
+                new ResolverChain([
+                    new AssociativeArrayResolver,
+                    new TypeHintContainerResolver($this->container),
+                ])
+            );
+        };
+
+        // Override the callback resolver with ours.
+        $app['callback_resolver'] = function ($app) {
+            return new CallbackResolver(
+                $app,
+                $app['phpdi.callable_resolver']
+            );
+        };
+    }
+
+    public function subscribe(\Pimple\Container $app, EventDispatcherInterface $dispatcher)
+    {
+        // Remove the Silex listeners first
+        $this->removeSubscriber($dispatcher, \Silex\EventListener\MiddlewareListener::class);
+        $this->removeSubscriber($dispatcher, \Silex\EventListener\ConverterListener::class);
+
+        // And register ours instead
+        $dispatcher->addSubscriber(new MiddlewareListener($app, $this->callbackInvoker));
+        $dispatcher->addSubscriber(new ConverterListener($app['routes'], $app['callback_resolver'], $this->callbackInvoker));
+    }
+
+    /**
+     * Removes an event subscriber by its class name.
+     *
+     * @param EventDispatcherInterface $dispatcher
+     * @param string                   $subscriberClass
+     */
+    private function removeSubscriber(EventDispatcherInterface $dispatcher, $subscriberClass)
+    {
+        if (!is_subclass_of($subscriberClass, EventSubscriberInterface::class, true)) {
+            throw new \InvalidArgumentException('$subscriberClass must implement ' . EventSubscriberInterface::class);
```

Comment:
Because `$subscriberClass` is the string class name and not an instance. Although I'm not sure it would be "bad" to use an instance here instead. But it's not really needed because the only method we call on the class is a static one.
Maybe we don't even need this check at all because it's a private method...